src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3214,6 +3216,12 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                     case t @ ClassDef(_, _, _, _) => t.symbol == cdef.symbol   // cdef ne t
                     case _ => false
                   }
+                case _ if sym.companionClass.isDerivedValueClass =>
+                  val csym = sym.companionClass
+                  stats.exists {
+                    case t @ ClassDef(_, _, _, _) => t.symbol eq csym
+                    case _ => false
+                  }
```

Comment:
Would it be too dry to refactor to `=> classInStats(_.symbol == sym.companionClass)` and the other case `classInStats(_.symbol == att.caseClass.symbol)` or what have you?

Reply:
Good call. I'm absorbing so much information at NEScala that I've got little chance of being (metaphorically) DRY.